### PR TITLE
cargo web prepare-emscripten

### PR DESCRIPTION
**Pull request by Nessex**
As per the discussion in https://github.com/koute/cargo-web/issues/22 this adds a basic command for installing emscripten separately to a build. The intention behind this is preparing the environment before a build for something like CI. As an example, I would use it in this repo http://github.com/Nessex/cargo-web-docker rather than building a placeholder project with no content. (See: https://github.com/Nessex/cargo-web-docker/blob/404c8ce5e3057a56fcc5b2832f9d78321c3761a9/Dockerfile#L11)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -308,10 +309,15 @@ fn main() {
...
 
+    let mut prepare_emscripten_subcommand =
+        SubCommand::with_name( "prepare-emscripten" )
+            .about( "Prepares and installs emscripten" );
+
     build_subcommand = add_shared_build_params( build_subcommand );
     test_subcommand = add_shared_build_params( test_subcommand );
     start_subcommand = add_shared_build_params( start_subcommand );
     deploy_subcommand = add_shared_build_params( deploy_subcommand );
+    prepare_emscripten_subcommand = add_shared_build_params( prepare_emscripten_subcommand );
```
Comment (koute) [inline src/main.rs]: This shouldn't have any shared build params attached to it as it doesn't use any of them anyway.

File: src/cmd_prepare_emscripten.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+use emscripten::initialize_emscripten;
+use error::Error;
+
+pub fn command_prepare_emscripten< 'a >() -> Result< (), Error > {
+    match initialize_emscripten( false, true ) {
+        None => return Err( Error::BuildError ),
+        Some( _emscripten ) => return Ok( () ),
```
Comment (koute) [inline src/cmd_prepare_emscripten.rs]: 1. Please add a new variant to the `Error` type, something like `Error::EmscriptenNotAvailable`. It's `description` can be something like `"prepackaged Emscripten is not available for this platform"`.
2. Return it on `None` case.

File: src/main.rs [hunk 3]
```diff
@@ -308,10 +309,15 @@ fn main() {
         SubCommand::with_name( "deploy" )
             .about( "Deploys your project so that its ready to be served statically" );
 
+    let mut prepare_emscripten_subcommand =
+        SubCommand::with_name( "prepare-emscripten" )
+            .about( "Prepares and installs emscripten" );
```
Comment (koute) [inline src/main.rs]: `"Fetches and installs prebuilt Emscripten packages"` would be a little better as a description, I think.

**Comment by Nessex** [comment]
Appreciate the feedback, will get back to that this week 👍

**Comment by koute** [comment]
LGTM; thanks!

**Pull request merged**